pkg/destroy/openstack/openstack.go

```diff
@@ -422,6 +452,28 @@ func deletePorts(opts *clientconfig.ClientOpts, listOpts ports.ListOpts, logger
 				}
 			}
 
+			// If there is a security group created by cloud-provider-openstack we should find it and delete it.
+			// We'll look through the ones on each of the ports and attempt to remove it from the port and delete it.
+			// Most of the time it's a conflict, but last port should be guaranteed to allow deletion.
+			// TODO(dulek): Currently this is the only way to do it and if delete fails there's no way to get back to
+			//              that SG. This is bad and we should make groups created by CPO tagged by cluster ID ASAP.
+			ports.Update(conn, port.ID, ports.UpdateOpts{
+				SecurityGroups: &[]string{}, // We can just detach all, we're deleting this port anyway.
+			})
+			for _, groupID := range port.SecurityGroups {
+				if group, ok := sgByID[groupID]; ok {
+					if cloudProviderSGNameRegexp.MatchString(group.Name) {
+						logger.Debugf("Deleting cloud-provider-openstack SG %q", groupID)
+						err := sg.Delete(conn, groupID).ExtractErr()
+						var err404 gophercloud.ErrDefault404
+						var err409 gophercloud.ErrDefault409
+						if !errors.As(err, &err404) && !errors.As(err, &err409) { // Ignore 404 Not Found and 409 Conflict
```

Comment:
If there was no error when deleting the sg, should we clean it up from the `sgByID` map? perhaps this way we avoid unnecessary calls to delete sgs when they're already gone

Reply:
Hm, not on 409, but on 404 for sure.